h/activity/query.py

```diff
@@ -123,11 +123,12 @@ def execute(request, query, page_size):
     # Add group information to buckets and present annotations
     for timeframe in result.timeframes:
         for bucket in timeframe.document_buckets.values():
+            bucket.presented_annotations = []
             for index, annotation in enumerate(bucket.annotations):
-                bucket.annotations[index] = {
+                bucket.presented_annotations.append({
```

Comment:
This is odd. I wonder why `index` and `enumerate()` were used here in the first place? As you've noticed it's not necessary to use `index`. I think you can actually change the `for index, annotation in enumerate(bucket.annotations):` to `for annotation in bucket.annotations:` as well?